src/pages/index.js

```diff
@@ -119,39 +99,91 @@ function closeModal(modal) {
   modal.classList.remove("modal_opened");
 }
 
-function closeModalOnEscape(event) {
-  if (event.key === "Escape") {
+function closeModalOnEscape(evt) {
+  if (evt.key === "Escape") {
     const modals = document.querySelectorAll(".modal_opened");
     modals.forEach(closeModal);
   }
 }
 
-function handleEditProfileFormSubmit(event) {
-  event.preventDefault();
-
-  profileName.textContent = profileNameInput.value;
-  profileDescription.textContent = profileDescriptionInput.value;
-
-  closeModal(editModal);
+function handleAvatarFormSubmit(evt) {
+  evt.preventDefault();
+  const saveBtnEl = evt.submitter;
+  const { avatar } = evt.target.elements;
+
+  setBtnText(saveBtnEl, false);
+  api
+    .editUserAvatar(avatar.value)
+    .then((data) => {
+      avatarEl.src = data.avatar;
+      closeModal(editAvatarModal);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => setBtnText(saveBtnEl, true));
 }
 
-function handleNewPostSubmit(event) {
-  event.preventDefault();
-  const link = newPostImageLinkInput.value;
-  const name = newPostNameInput.value;
-  event.target.reset();
-
-  const cardEl = getCardElement({
-    link,
-    name,
-  });
+function handleEditProfileFormSubmit(evt) {
+  evt.preventDefault();
+  const saveBtnEl = evt.submitter;
+  const { name, about } = evt.target.elements;
+
+  setBtnText(saveBtnEl, false);
+  api
+    .editUserInfo({
+      name: name.value,
+      about: about.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => setBtnText(saveBtnEl, true));
+}
 
-  cardsList.prepend(cardEl);
-  closeModal(newPostModal);
+function handleNewPostSubmit(evt) {
+  evt.preventDefault();
+  const addPostBtnEl = evt.submitter;
+  const { name, link } = evt.target.elements;
+
+  setBtnText(addPostBtnEl, true, "Add post", "Adding post...");
+  api
+    .addPost({
+      name: name.value,
+      link: link.value,
+    })
+    .then((cardData) => {
+      const cardEl = getCardElement(cardData);
+      cardList.prepend(cardEl);
```

Comment:
**Could be improved**

You can make a universal function for adding a card into the section using different methods: `append`, `prepend`, or anything else:

```js

// The function accepts a card object and a method of adding to the section
// The method is initially `prepend`, but you can pass `append` 
function renderCard(item, method = "prepend") {
  
  const cardElement = getCardElement(item);
  // Add the card into the section using the method
  cardList[ method ](cardElement);
}



```

Reply:
Don't really understands why we need this. If I need append - I use dedicated method and same with prepend.
`cardList.append(cardEl)` and `renderCard(cardEl, "append")` do not differs so much to make a separate method.
Also renderCard - could mean a lot of things, not only just appending or prepending an element.

Reply:
> Don't really understands why we need this


You have thse 2 lines of code

```js

      const cardEl = getCardElement(cardData);
      cardList.prepend(cardEl);
```


And you have the other 2 lines of code


```js
        const cardEl = getCardElement(cardData);
        cardList.append(cardEl);
```


So, you repeat `const cardEl = getCardElement(cardData);` in each of them

Thus, to avoid repeating you need this `renderCard`